### PR TITLE
Glimpse2 phase: move to nf-test

### PR DESCRIPTION
**Pull request by LouisLeNezet**
Add nf-test to glimpse2/phase

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/glimpse2/phase/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                // File has a timestamp in it and is in binary format, so we can only check the name
+                { assert file(process.out.phased_variants[0][1]).name == "input_chr21_16650000-16750000.bcf" },
+                { assert snapshot(process.out.version).match("VCF")}
```
Comment (mashehu) [inline modules/nf-core/glimpse2/phase/tests/main.nf.test:50]: maybe stupid question, but shouldn't the version snapshot be the same between the different tests?

Comment (LouisLeNezet) [reply modules/nf-core/glimpse2/phase/tests/main.nf.test:50]: It should effectively, so if I understood correctly the string in the match() function should be the same for it to test that all are identical ?

**Comment by LouisLeNezet** [comment]
> please remove the old pytest tests and their tags.yml entry 🙂

Apparently it was already deleted, I didn't find it in `pytest_modules.yml`

File: modules/nf-core/glimpse2/phase/tests/main.nf.test (hunk 1 above)
Comment (LouisLeNezet) [reply modules/nf-core/glimpse2/phase/tests/main.nf.test:50]: This doesn't seems to work ^^
`A snapshot with id 'version' already exists. Snapshot ids have to be unique, and a test can only have one unnamed snapshot.`

**Review by mashehu — APPROVED** [review]

File: modules/nf-core/glimpse2/phase/tests/main.nf.test (hunk 1 above)
Comment (LouisLeNezet) [reply modules/nf-core/glimpse2/phase/tests/main.nf.test:50]: There were a small mistakes `version` instead of `versions`

**Pull request merged**